### PR TITLE
Login and signup

### PR DESCRIPTION
**Pull request by HeyNitin**
Added login and signup pages

https://deploy-preview-2--strivesocial.netlify.app/

### CODE REVIEW

File: src/components/auth/requireAuth.tsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
-// import { useLocation } from "react-router-dom";
-// import { useAuth } from "contexts/authContext/authContext";
 import { ReactNode } from "react";
+import { useLocation, Navigate } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "appRedux/hooks";
+import { setToken } from "appRedux/userSlice";
 
 const RequireAuth = ({ children }: { children: ReactNode }): JSX.Element => {
-	// const { token } = useAuth();
-	// const res = localStorage.getItem("token");
```
Comment (vermasavita) [inline src/components/auth/requireAuth.tsx:7]: Remove these comments

File: src/components/auth/requireAuth.tsx [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+	const location = useLocation();
+	const Dispatch = useAppDispatch();
+
+	res && Dispatch(setToken(res));
 
 	return (
-		<>{children}</>
-		// <>
-		// 	{res || token ? (
-		// 		children
```
Comment (vermasavita) [inline src/components/auth/requireAuth.tsx:14]: Remove these comments

File: src/components/auth/requireNoAuth.tsx [hunk 3]
```diff
@@ -1,26 +1,30 @@
-// import { useLocation } from "react-router-dom";
-// import { useAuth } from "contexts/authContext/authContext";
 import { ReactNode } from "react";
+import { useLocation, Navigate } from "react-router-dom";
+import { useAppSelector, useAppDispatch } from "appRedux/hooks";
+import { setToken } from "appRedux/userSlice";
 
 const RequireNoAuth = ({ children }: { children: ReactNode }): JSX.Element => {
-	// const { token } = useAuth();
-	// const res = localStorage.getItem("token");
```
Comment (vermasavita) [inline src/components/auth/requireNoAuth.tsx:7]: Remove it

File: src/components/auth/requireNoAuth.tsx [hunk 4]
```diff
@@ -1,26 +1,30 @@
...
+	res && Dispatch(setToken(res));
 
-	// const state = location?.state as { from: { pathname: string } };
 	return (
-		<>{children}</>
-		// <>
-		// 	{res || token ? (
-		// 		<Navigate
-		// 			to={state?.from?.pathname || "/"}
-		// 			state={{ from: location }}
```
Comment (vermasavita) [inline src/components/auth/requireNoAuth.tsx:17]: Remove comments please

**Review by vermasavita — COMMENTED** [review]
It's impressive to see that you have validated your form. Great work there.

File: src/pages/login/login.tsx [hunk 5]
```diff
@@ -1,5 +1,148 @@
...
+		e.preventDefault();
+		if (state.userName !== "" && state.password !== "") {
+			try {
+				const { data } = await axios.post("/api/auth/login", {
+					username: state.userName.toLowerCase(),
+					password: state.password,
+				});
+				Dispatch(setToken(data.encodedToken));
+				switch (state.rememberMe) {
+					case true:
```
Comment (vermasavita) [inline src/pages/login/login.tsx:36]: Just format your data so that will be easy to read

**Review by vermasavita — APPROVED** [review]
Approving.

**Review by gauravsinhaweb — APPROVED** [review]
couldn't find any major improvement, you can merge it.

**Pull request merged**